nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-providers/src/main/java/org/apache/nifi/parameter/aws/AwsSecretsManagerParameterProvider.java

```diff
@@ -70,12 +71,26 @@
 @CapabilityDescription("Fetches parameters from AWS SecretsManager.  Each secret becomes a Parameter group, which can map to a Parameter Context, with " +
         "key/value pairs in the secret mapping to Parameters in the group.")
 public class AwsSecretsManagerParameterProvider extends AbstractParameterProvider implements VerifiableParameterProvider {
+    public static final String ENUMERATED_STRATEGY = "Enumerated";
+    public static final String PATTERN_STRATEGY = "Pattern";
+    public static final PropertyDescriptor SECRET_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .name("secret-listing-strategy")
+            .displayName("Secret Listing Strategy")
+            .description("Strategy to use when listing secrets. 'Pattern' strategy treats Secret Name Pattern as a regular expression and fetches  " +
+                    "all secrets whose names match the pattern. 'Pattern' strategy requires ListSecrets and GetSecretValue permissions. "+
+                    "'Enumerated' strategy treats Secret Name Pattern as a coma-separated list and fetches all secrets whose names are in the list. " +
+                    "'Enumerated' strategy requires only GetSecretValue permission.")
```

Comment:
Instead of describing the option in the property description, each value should be an instance of `AllowableValue` and include a description property.

Reply:
I agree with David to move the description of the options to each option itself respectively.

An alternative approach to using `AllowableValue` is the declaration of an `enum` that implements the `DescribedValue` interface. An example can be found [here](https://github.com/apache/nifi/blob/92f571940edbb65c423d47b6881db638572fe466/nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java#L268-L300). 
Due to improved type safety at the use site, this approach allows for exhaustive switch expressions, as can be seen [here](https://github.com/apache/nifi/blob/92f571940edbb65c423d47b6881db638572fe466/nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java#L214-L218) and [here](https://github.com/apache/nifi/blob/92f571940edbb65c423d47b6881db638572fe466/nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java#L181-L185) for example. 

The advantage I see in this approach is that the compiler enforces to handle all options on all use sites, especially as new options are added. 

Regardless of the approach you take, moving the descriptions to the options is sensible.